### DISCUSSION TITLE
Notepad, Divide Concepts and questions by age group, Add a Math concepts page

### DISCUSSION THREAD

**Discussion by Jiya55** [prakhartiwari0/Arito#84] [Ideas]
Hello,
I was looking through the website and the codes and I thought of a few ideas which could be implemented. I want to contribute to this project too so I could try to help with the ideas. I have also made a python tkinter GUI project which was to practice the tables of a few common numbers. My ideas are -

1. To add a notepad to do calculation on the laptop itself- I noticed that websites like khan academy have a drawing tool while solving questions with which people can calculate without having a pen paper. So i was thinking if that could be implemented.
2. I was seeing another post in Ideas discussion, which about implementing square and cube roots. So I was thinking if the concepts could be divided by grade level or age groups.
3. I think it would be helpful if there could be a page which teaches the concepts to the children. These could be the basic concepts for which the questions already exist and the more advanced concepts like how to find the square and cube roots of large numbers with long division.

I can help with the point 3 and try to help with point 1,2 and adding more types of questions on the website.Thank you for reading

**Comment by 3xp11i** [comment]
These ideas are cool, but I am not sure how hard they will be to implement. This is my take on your points ->

1. I am aware of those digital tools. But personally, they are useless for me because first of all, on a laptop or computer, we use a mouse or touchpad which is not good enough to use for something like that. Also, we need not make kids stick to the device, they should have a habit of using pen and paper and not go completely digital (just an opinion). Because in real life, I don't think in most of the schools there is any use of digital tools like these. I would love to have more points in favour of this to think forward in implementing this.

2. Yeah we need to introduce different kinds of questions gradually so that we can serve more children. But it is a little difficult process as we need to rethink how the entire app should behave, and what should the functionality be like. If you have any proper plan on how to implement them, I would love to know that!

3. This app is built for practising and not teaching, so we should not pivot from our main path. The concepts are not hard and they are taught in schools or homes. Children can learn online from thousands of resources already available, so ours wouldn't make much difference and it will only make the app more complex. I hope you understand my point.

Thanks for your time

**Reply by Jiya55** [reply]
yeah, even i agree on point 1 as i use my laptop touchpad to write on khan academy and have trouble writing legitimately. And a lot of use of digital devices is anyways bad(coming from a person who spends her whole day on the laptop). 
as for point 2, yeah It will take a lot of remodeling honestly if we were to implement it. But i think it is better if it is done during the early stages only when there is less code. One way of implementing could be to make a navigation bar as the top of the pages. In the navigation bar, the age-groups/grade levels could be implemented kind of like how khan academy has a grade level wise divided navigation bar. 

I guess if the the aim of the website is only for practicing as of now then sure. Though I do think if would make a difference as the children can only use one website to learn and practice concepts and it wold only be a little complicated to code.

**Reply by 3xp11i** [reply]
So we agree that point 1 is not much of a requirement. 
**Point 2** is something I need to think about, I will try to plan and design this newer version and will share all those ideas in this discussion. And people who are interested in the development of this project can share their ideas on how that can be implemented, or simply give feedback on my design. I would encourage you to try to design what you are thinking, and see how it works. It will be a collaborative effort to make this app upgrade to the next level.

**Point 3:** My main question is how and why?
Why explain so simple concepts like adding and subtracting, or even square roots and other things that may come? According to me, the app will lose its main purpose if we do that because it's the duty of the school, or guardians to make their children learn the concepts. We will serve the purpose of practising the skills rather than building them up. I know it will not be much complicated (if you aren't thinking of it to be interactive lol that would be insane) because it will be just content with text and images. But still, why take this step of teaching such basic concepts which they will obviously learn from outside?

**Reply by 3xp11i** [reply]
(Personal ideas)

Actually, I did think to make it a complete mathematics app where people could learn and practice their skills, but those ideas are better to be implemented in the future, this app is having a very niche purpose and is programmed using basic concepts of Vanilla JS (so that beginners can contribute to it). We can have another project for interactive learning of mathematics and that would also be an amazing project and will surely be very useful.

**Reply by Jiya55** [reply]
for point 2, yeah, even i will think more about how to implement it.
Point 3, i did actually think of making it interactive but i was kind of doubting whether i had enough skills to try to implement that as my JS is very weak if i can be honest. and Yeah i guess another project could be made for interactive learning as i have barely seen any website which actually teacher interactively and arent just articles and video lesson. If you do take that forward, i would love to be a part of it.

**Reply by 3xp11i** [reply]
Don't doubt your skills, let me share with you a life lesson, we learn the best when we need it. In this case, we will need to enhance our skills to make that app, and so we will!
And yeah I am gonna launch more cool projects (including that interactive learning app) in the future, I will share the updates on my Twitter. You will know when it gets launched. But for now, let's focus on upgrading Arito and introducing new features to it.

A lot of new contributors are coming and they will also be helping us, open source is great for this reason only, **we learn, we code, we collaborate, we connect, and We Win!**
